v2/src/core/router/helpers.ts

```diff
@@ -1,22 +1,24 @@
 import { RouteLocationNormalized, NavigationGuardNext } from 'vue-router'
 import { ADDRESS_ROUTE_QUERY } from './routesNames'
-export const addressTokensRouteGuard = async (to: RouteLocationNormalized, from: RouteLocationNormalized, next: NavigationGuardNext) => {
-    // check if TO query exhists and valid
-    if (!to.query.t) {
-        // redirect the user to first tab
-        to.query.t = ADDRESS_ROUTE_QUERY.Q_TOKENS[0]
-        next(to)
-    } else {
-        next()
+
+export const addressRouteGuard = (type: string) => {
+    let tab: string
```

Comment:
please use the ADDRESS_ROUTE_QUERY variable instead of the string whenever you call the function then you don't have to make the variable tab OR make a new enum for the 'tokens' and 'miner'